js_modules/dagster-ui/packages/ui-core/src/search/useIndexedDBCachedQuery.tsx

```diff
@@ -0,0 +1,67 @@
+import {DocumentNode, OperationVariables, useApolloClient} from '@apollo/client';
+import {cache} from 'idb-lru-cache';
+import React from 'react';
+
+const ONE_WEEK = 1000 * 60 * 60 * 24 * 7;
+
+/**
+ * Returns data from the indexedDB cache initially while loading is true.
+ * Fetches data from the network/cache initially and does not receive any updates afterwards
+ */
+export function useIndexedDBCachedQuery<TQuery, TVariables extends OperationVariables>({
```

Comment:
It might be useful to add an option where we DO subscribe to the cache for updates, for that version we'd use `useLazyQuery` instead of calling `client.query`